tests/core/schedulers_test.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+import pytest
+
+
+@pytest.mark.parametrize(
+    ["max_trials", "success"],
+    [(4, True), (5, False)],
+)
+def test_retry_decorator_max_trial(max_trials: int, success: bool):
+    import time
+
+    from beamlime.core.schedulers import retry
+
+    interval = 0.2
+    error_throw_ranges = iter(range(max_trials - int(success) * 1))
+    context = {"run_counts": 0}
+
+    @retry(TimeoutError, max_trials=max_trials, interval=interval)
+    def something_takes_time():
+        context["run_counts"] += 1
+        for _ in error_throw_ranges:
+            raise TimeoutError
+        return True
+
+    start_time = time.time()
+    if not success:
+        with pytest.raises(TimeoutError):
+            something_takes_time()
+    else:
+        assert something_takes_time()
+
+    time_consumed = time.time() - start_time
+    expected_consumed_time = (max_trials - 1) * interval
+    assert time_consumed > expected_consumed_time
+    assert (
+        time_consumed - expected_consumed_time
+    ) < 0.01  # Expected delta for several trials.
```

Comment:
If possible, I'd make this larger, 10 ms may make this test flaky.

Reply:
Yes... it was very close to the actual difference.
I increased it to `0.05` instead of `0.01`.
I'll add a brief benchmark in the documentation as well. #76